packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,314 @@
+    }
+    this.artifactBucket = props.artifactBucket ?? new s3.Bucket(this, 'ServiceBucket');
+
+    // Created role will need these policies to run the Canary.
+    // These are the necessary permissions as listed here:
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-synthetics-canary.html
+    const policy = new iam.PolicyDocument({
+      statements: [
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: ['s3:ListAllMyBuckets'],
+        }),
+        new iam.PolicyStatement({
+          resources: [`${this.artifactBucket.bucketArn}`],
+          actions: ['s3:PutObject', 's3:GetBucketLocation'],
+        }),
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: ['cloudwatch:PutMetricData'],
+          conditions: {StringEquals: {'cloudwatch:namespace': 'CloudWatchSynthetics'}},
+        }),
+        new iam.PolicyStatement({
+          resources: ['arn:aws:logs:::*'],
+          actions: ['logs:CreateLogStream', 'logs:CreateLogGroup', 'logs:PutLogEvents'],
+        }),
+      ],
+    });
+
+    this.role = props.role ?? new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+      inlinePolicies: {
+        canaryPolicy: policy,
+      },
+    });
+
+    const frequency = props.frequency ?? cdk.Duration.minutes(5);
+    this.validateFrequency(frequency);
+    const memory = props.memorySize?.toMebibytes() ?? 960;
+    this.validateMemory(memory);
+    // Timeout defaults to the frequency or 900, whichever is smaller. It also may not be greater than the frequency.
+    var timeout = props.timeout ?? cdk.Duration.seconds(Math.min(frequency.toSeconds(), 900));
+    timeout = cdk.Duration.seconds(Math.min(timeout.toSeconds(), frequency.toSeconds()));
+
+    const resource: CfnCanary = new CfnCanary(this, 'Resource', {
+      artifactS3Location: this.artifactBucket.s3UrlForObject(),
+      executionRoleArn: this.role.roleArn,
+      startCanaryAfterCreation: props.enable ?? true,
+      runtimeVersion: 'syn-1.0',
+      name: this.physicalName,
+      runConfig: {
+        memoryInMb: memory,
+        timeoutInSeconds: timeout.toSeconds(),
+      },
+      schedule: {
+        durationInSeconds: props.timeToLive ? String(props.timeToLive.toSeconds()) : '0' ,
+        expression: this.createExpression(frequency),
+      },
+      failureRetentionPeriod: props.failureRetentionPeriod?.toDays(),
+      successRetentionPeriod: props.successRetentionPeriod?.toDays(),
+      // 🚧 TODO: implement
+      code: {
+        handler: 'index.handler',
+        script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+      },
+    });
+
+    this.canaryId = resource.attrId;
+    this.canaryState = resource.attrState;
+    this.canaryName = this.getResourceNameAttribute(resource.ref);
+  }
+
+  /**
+   * Returns a new metric for the canary
+   *
+   * @default avg over 5 minutes
```

Comment:
Here, if you specify `statistic` in `options`, it will overwrite the default of `avg`. I've added a test case to demonstrate this, and this pattern is what I observed in other metric methods like that in `state-machine.ts` in `stepfunctions`.

Reply:
Yes it will work: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#object-spread-and-rest

This is because you passed the `...options` as an Object rests. Note that the order matters here:

```
{
      ...options,
      metricName,
      namespace: 'CloudWatchSynthetics',
      dimensions: { CanaryName: this.canaryName },
      statistic: 'avg',
}
```
The value passed to the `statistic` property would have override the `options` property.

Reply:
Yes -- I wrote the same thing for a small PR to expose `metric()` in `metricFilter`, and I was very careful with this because it does seem odd.